src/games/kalambury/sidebar/Player.js

```diff
@@ -15,8 +17,10 @@ const Player = ({
   handleGuessClick,
 }) => {
   const { t } = useTranslation("kalambury");
+  const { getNickName } = useContext(UserContext);
+  const { data } = useQuery(["user-nickname", name], (key, id) => getNickName(id));
```

Comment:
I think we can extract this whole thing as a custom hook like `useNickname(uid)`

or process those nicknames when we fetch list of rooms and assign them to each player, so other componets doesnt need to worry about this